### PR TITLE
feat(rpc): allow configure WS compression

### PR DESCRIPTION
**Pull request by mask-pp**
Enable ws compression by interface.
Enable it by call `Server.SetCompressionLevel(flate.BestCompression)`.

### CODE REVIEW

**Comment by Thegaram** [comment]
I just realized that the HTTP JSON-RPC endpoint enabled gzip by default:

```
> curl -H 'Content-Type: application/json' -H 'Accept: application/gzip' -H 'Accept-Encoding: gzip' -d '{"jsonrpc":"2.0","id":"id","method":"eth_blockNumber"}' https://prealpha-rpc.scroll.io/l2 --output -
```

With your change, how can the WS client enable/disable compression? I usually test using `websocat -v wss://prealpha-rpc.scroll.io/l2/ws` but it seems `websocat` has no compression option.

File: rpc/websocket.go [hunk 1]
```diff
@@ -50,10 +50,11 @@ var wsBufferPool = new(sync.Pool)
 // To allow connections with any origin, pass "*".
 func (s *Server) WebsocketHandler(allowedOrigins []string) http.Handler {
```
Comment (0xmountaintop) [inline rpc/websocket.go:52]: if we change to `WebsocketHandler(enableCompression bool, allowedOrigins []string) http.Handler {` we can acheive only compress between coordinator&rollers

Comment (mask-pp) [reply rpc/websocket.go:52]: Use `SetCompressionLevel` function can let it work.

**Comment by mask-pp** [comment]
> 

Good question, i have added `SetCompressionLevel` function, in order to set compression level when necessary.

**Review by 0xmountaintop — APPROVED** [review]

**Review by colinlyguo — APPROVED** [review]
LGTM.

**Review by Thegaram — APPROVED** [review]

**Pull request merged**